otherlibs/unix/getgr.c

```diff
@@ -43,31 +46,77 @@ static value alloc_group_entry(struct group *entry)
 
 CAMLprim value caml_unix_getgrnam(value name)
 {
-  struct group * entry;
+  value res;
+  struct group *resultp;
   if (! caml_string_is_c_safe(name)) caml_raise_not_found();
+
+#ifdef HAVE_GETGRNAM_R
+  long int initlen = sysconf(_SC_GETGR_R_SIZE_MAX);
+  size_t len = initlen == -1 ? /* default */ 1024 : (size_t) initlen;
```

Comment:
It doesn't hurt to write
```suggestion
  size_t len = initlen <= 0 ? /* default */ 1024 : (size_t) initlen;
```
which will appease static analyzers complaining about the risk of casting a negative `long` value into an unsigned `size_t` value.

Reply:
Thanks!